server/middleware/setupFeatureFlags.ts

```diff
@@ -30,9 +31,22 @@ export const featureFlagHandler = (featureFlagService: FeatureFlagService): Requ
 }
 
 export const retrieveFlags = (featureFlagService: FeatureFlagService) => {
-  featureFlagsToUse.forEach(async flag => {
-    const flagBool = await featureFlagService.getBooleanFlag(flag)
+  return Promise.all(
+    featureFlagsToUse.map(async flag => {
+      const flagBool = await featureFlagService.getBooleanFlag(flag)
+      process.env[flag] = flagBool.toString()
+    }),
+  )
+}
 
-    process.env[flag] = flagBool.toString()
-  })
+export const retrieveFlag = async (flag: FeatureFlag, featureFlagService: FeatureFlagService) => {
```

Comment:
may be this can be simplified
export const retrieveFlag = async (flag: FeatureFlag, featureFlagService: FeatureFlagService) => {
  let flagValue = false
  try {
    flagValue = await featureFlagService.getBooleanFlag(flag)
  } catch (error) {
    logger.error(`Error retrieving feature flag ${flag}`, error)
  }
  return flagValue
}

Reply:
yeah I think that is a good idea